realm/realm-library/src/main/cpp/java_accessor.hpp

```diff
@@ -154,6 +154,11 @@ class JObjectArrayAccessor {
     jsize m_size;
 };
 
+template<>
+inline JStringAccessor JObjectArrayAccessor<JStringAccessor,jstring>::operator[](const int index) const noexcept {
```

Comment:
Is this being used anywhere?

Reply:
It is used by https://github.com/realm/realm-java/pull/7232/files#diff-8905bc013670a06267a7fd4be2a36b6698ed38d2d3f442c88d1223f57e8e8772R107. The problem is the reference generated by the below `GetObjectArrayElement`, which is immediately passed into the JStringAccess. I did not see any other reasonable easy way to grab it and delete without doing it through the `JStringAccessor`, without having to verify all usages of JStringAccessor through out our code. It is actually immediately releasable as the data is copied out in the constructor, so basically just propagating the `owned` argument to the inner temporary `JStringCharsAccessor`. Maybe `delete_jstring_ref` would be more appropriate.